sway-ir/src/instruction.rs

```diff
@@ -165,6 +165,167 @@ pub enum Register {
 }
 
 impl Instruction {
+    /// Return a string representation of type, used for debug printing.
+    pub fn as_debug_string(&self, context: &Context) -> String {
+        match self {
```

Comment:
I thought about that, but the format of this `as_debug_string` is in the style of `fmt::Debug`, the idea was to hopefully keep a consistent style of debugging output, whereas the printer outputs the information in a different style.

Anyway this debug stuff is not really necessary for this PR, so I can take it out and can send a separate PR adding the debugging-specific code.

Reply:
I still disagree here.  The serialised IR should contain everything about the IR, particularly since we parse it back into IR from this format.  In that regard, it should be fine for debug output and duplicating all the effort is wasteful and harder to maintain.

It'd be nice to have `Debug` (and `dbg!()`) or `Display` but it isn't possible due to needing the `Context` for nice output.